### PR TITLE
feat: expose widget methods

### PR DESCRIPTION
**Pull request by RomanistNLX**

### CODE REVIEW

File: packages/widget/src/example/index.tsx [hunk 1]
```diff
@@ -25,4 +25,13 @@ const props: Props = {
   },
 };
 
-render(<Widget {...props} />, document.querySelector("#app"));
```
Comment (RomanistNLX) [inline packages/widget/src/example/index.tsx:28]: this is just for test

File: packages/widget/src/utils.ts [hunk 2]
```diff
@@ -24,3 +27,57 @@ const groupWhileHelper = <T>(
   }
   return groupWhileHelper([...currentGroup, head], tail, breakGroup);
 };
+
+// https://stackoverflow.com/a/55243880/11698825
+
+export class ReactHandler<PropsType> {
```
Comment (RomanistNLX) [inline packages/widget/src/utils.ts:33]: I think it's a very useful finding from my side, we could use it in other places too

File: packages/widget/src/utils.ts [hunk 3]
```diff
@@ -24,3 +27,57 @@ const groupWhileHelper = <T>(
...
+
+  private container: any;
+  private props: any;
+
+
+  constructor(private component: any) {
```
Comment (RomanistNLX) [inline packages/widget/src/utils.ts:39]: ts went crazy for me here and didn't want to accept react component type. We might need to update version or something

File: packages/widget/src/index.tsx [hunk 4]
```diff
@@ -17,23 +17,34 @@ import { CloseIcon, ChatIcon, AirplaneIcon, DownloadIcon } from "./icons";
...
   document.body.appendChild(node);
-  render(<Widget {...props} />, node);
+  reactHandler.mount(node, props);
+
   return {
     teardown: () => {
       unmountComponentAtNode(node);
     },
+    expand: () => {
+      reactHandler.updateProps({ ...props, initiallyExpanded: true });
```
Comment (peterszerzo) [inline packages/widget/src/index.tsx:43]: The `initiallyExpanded` prop is intended to control whether the widget is expanded at the very beginning at the time it is mounted.

File: packages/widget/src/index.tsx [hunk 5]
```diff
@@ -150,7 +161,9 @@ export const Widget: React.FunctionComponent<Props> = (props) => {
   }, [chat.conversationHandler]);
 
   // Expanded state
-
+  useEffect(() => {
+    setExpanded(Boolean(props.initiallyExpanded));
+  }, [Boolean(props.initiallyExpanded)])
```
Comment (peterszerzo) [inline packages/widget/src/index.tsx:166]: I have implementation clarity concerns here per https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state.

**Comment by peterszerzo** [comment]
@RomanistNLX I thought it through and I still don't think this solution is the right way to go. Happy to discuss today if you have any questions or if you would still like to make a case for it. Thank you for the effort in any case, this must not have been easy to get working!

**Pull request closed without merging**